File: examples/node/s3-delete.js
// Elimina un objeto de S3 (y opcionalmente el bucket si queda vacío).
// Uso: node s3-delete.js <bucket> <clave> [--bucket-tambien]
const AWS = require('aws-sdk');

const s3 = new AWS.S3({
  endpoint: 'http://localhost:4566',
  region: 'us-east-1',
  credentials: { accessKeyId: 'test', secretAccessKey: 'test' },
  s3ForcePathStyle: true,
});

const [, , bucket, key, flag] = process.argv;

async function main() {
  if (!bucket || !key) {
    throw new Error('Uso: node s3-delete.js <bucket> <clave> [--bucket-tambien]');
  }
  await s3.deleteObject({ Bucket: bucket, Key: key }).promise();
  console.log(`Objeto eliminado: s3://${bucket}/${key}`);

  if (flag === '--bucket-tambien') {
    await s3.deleteBucket({ Bucket: bucket }).promise();
    console.log(`Bucket eliminado: ${bucket}`);
  }
}

main().catch((error) => {
  if (error.code === 'BucketNotEmpty') {
    console.error('El bucket todavía tiene objetos. Elimínalos antes de borrar el bucket.');
  } else {
    console.error('Error eliminando:', error.message);
  }
  process.exit(1);
});

File: examples/python/s3_delete.py
"""Elimina un objeto de S3 (y opcionalmente el bucket si queda vacío).

Uso: python3 s3_delete.py <bucket> <clave> [--bucket-tambien]
"""
import sys

import boto3
from botocore.exceptions import ClientError

s3 = boto3.client(
    "s3",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 3:
        raise SystemExit("Uso: python3 s3_delete.py <bucket> <clave> [--bucket-tambien]")
    bucket, key = sys.argv[1], sys.argv[2]
    borrar_bucket = "--bucket-tambien" in sys.argv[3:]

    s3.delete_object(Bucket=bucket, Key=key)
    print(f"Objeto eliminado: s3://{bucket}/{key}")

    if borrar_bucket:
        try:
            s3.delete_bucket(Bucket=bucket)
            print(f"Bucket eliminado: {bucket}")
        except ClientError as error:
            if error.response["Error"]["Code"] == "BucketNotEmpty":
                raise SystemExit("El bucket todavía tiene objetos. Elimínalos antes de borrar el bucket.")
            raise


if __name__ == "__main__":
    main()
